### ISSUE TITLE
meshroom_batch change FeatureExtraction:describerTypes syntax

### ISSUE THREAD

**Issue by sbolten** [alicevision/Meshroom#1577]
Using the command line tool "meshroom_batch" I am able to override parameters like the "forceCpuExtraction" successfully like this:

`./meshroom_batch -i /input -o /output --paramOverrides FeatureExtraction:forceCpuExtraction=0`

If I wantet o override the "describerTypes" to sift + akaze (for example) as well I tried something like this:

`./meshroom_batch -i /input -o /output --paramOverrides FeatureExtraction:forceCpuExtraction=0 FeatureExtraction:describerTypes=sift,akaze`

This does not work. The error message I get is the following:

`ValueError: ChoiceParam value "s" is not in "['sift', 'sift_float', 'sift_upright', 'dspsift', 'akaze', 'akaze_liop', 'akaze_mldb', 'cctag3', 'cctag4', 'sift_ocv', 'akaze_ocv']".`

So obviously the syntax I use to assign the two types via command line is incorrect.
Any help on the correct syntax would be appreciated.

I am running on Linux (arch) with the latest version of meshroom (2021.1.0)

**Comment by simogasp** [comment]
from here
https://github.com/alicevision/meshroom/blob/36896c1238a2b72e0bbda394472246fd5270a8af/bin/meshroom_batch#L177-L198

it seems the value is not parsed but just assigned. I don't know if it is possible to use the python syntax directly to specify the value, like ```['sift','akaze']```

**Comment by sbolten** [comment]
Thx for the suggestion but unfortunately that does not solve my problem. The error message is similiar to what I wrote in my original message:

```
ValueError: ChoiceParam value "[" is not in "['sift', 'sift_float', 'sift_upright', 'dspsift', 'akaze', 'akaze_liop', 'a
kaze_mldb', 'cctag3', 'cctag4', 'sift_ocv', 'akaze_ocv']".
```
So it only takes the first character from the command line - in this case the "[" (square bracket)

Any other suggestions around?

Regards, Stephan

**Comment by sarveshkoyande** [comment]
Did you manage to find a work around?
I'm facing the same issue, and it seems there is no documentation around this.

Why does paramOverrides work for some instances but here it does not?

**Comment by sbolten** [comment]
Sorry. Did not find a workaround for this :-(Regards,StephanAm 13.01.2022 10:34 schrieb sarveshkoyande ***@***.***>:
Did you manage to find a work around?
I'm facing the same issue, and it seems there is no documentation around this.
Why does paramOverrides work for some instances but here it does not?

**Comment by yzho0907** [comment]
> ChoiceParam value

Right, simply u could try this: value = [value], might save the day

**Comment by natowi** [comment]
How about using a .mg file with the settings you need as a workaround?

**Comment by Dev-hestabit** [comment]
Anyone solved this issue ?

**Comment by stuardo-lucho** [comment]
anyone?

**Comment by cbentejac** [comment]
Hi!

This issue has been fixed by #1811 (in relation to #1797, which was a duplicate of this issue). The fix is available on the develop branch of Meshroom.
